core/node/state_keeper/src/io/tests/mod.rs

```diff
@@ -644,3 +666,121 @@ async fn insert_unsealed_batch_on_init(commitment_mode: L1BatchCommitmentMode) {
     assert_eq!(l1_batch_params.fee_input, fee_input);
     assert_eq!(l1_batch_params.first_l2_block.timestamp, 2);
 }
+
+#[test_casing(2, COMMITMENT_MODES)]
+#[tokio::test]
+async fn test_mempool_with_timestamp_assertion(commitment_mode: L1BatchCommitmentMode) {
+    let connection_pool = ConnectionPool::<Core>::constrained_test_pool(2).await;
+    let tester = Tester::new(commitment_mode);
+    let mut storage = connection_pool.connection().await.unwrap();
+
+    tester.genesis(&connection_pool).await;
+
+    // Insert a sealed batch so there will be a `prev_l1_batch_state_root`.
+    // These gas values are random and don't matter for filter calculation.
+    let tx_result = tester
+        .insert_l2_block(&connection_pool, 1, 5, BatchFeeInput::l1_pegged(55, 555))
+        .await;
+    tester
+        .insert_sealed_batch(&connection_pool, 1, &[tx_result])
+        .await;
+
+    // Create a copy of the tx filter that the mempool will use.
+    let want_filter = l2_tx_filter(
+        &tester.create_batch_fee_input_provider().await,
+        ProtocolVersionId::latest().into(),
+    )
+    .await
+    .unwrap();
+
+    // Create a mempool without pending batch and ensure that filter is not initialized just yet.
+    let (mut mempool, mut guard) = tester.create_test_mempool_io(connection_pool).await;
+    mempool.initialize().await.unwrap();
+    assert_eq!(mempool.filter(), &L2TxFilter::default());
+
+    let system_time = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("Time went backwards")
+        .as_secs() as i64;
+
+    // inserting 3 transactions - a good one, sandwiched in between two bad ones. The good one should
+    // be returned by wait_for_next_tx, while two bad ones should be rejected.
+    #[allow(deprecated)]
+    let rejected_tx_1 = tester.insert_tx(
+        &mut guard,
+        want_filter.fee_per_gas,
+        want_filter.gas_per_pubdata,
+        TransactionTimeRangeConstraint {
+            timestamp_asserter_range: Some((system_time - 20000, system_time - 10000)),
+        },
+    );
+    #[allow(deprecated)]
+    let expected_tx = tester.insert_tx(
+        &mut guard,
+        want_filter.fee_per_gas,
+        want_filter.gas_per_pubdata,
+        TransactionTimeRangeConstraint {
+            timestamp_asserter_range: Some((system_time - 1000, system_time + 1000)),
+        },
+    );
+    #[allow(deprecated)]
```

Comment:
Dumb question: What's deprecated in this statement? It seems to work fine if the attribute is removed.

Reply:
Thanks, good spot - it's a leftover from previous iterations where I used `NaiveDateTime` type